### PR TITLE
NewsDownloader: Keep nil reference from being passed to UIManager  

### PR DESCRIPTION
**Pull request by roygbyte**
*I can't believe I opened a PR and it didn't include unrelated commits.*

Can't reproduce, but most likely this is a fix for #9690

- - -
This change is [Reviewable](https://reviewable.io/reviews/koreader/koreader/9697)

### CODE REVIEW

File: plugins/newsdownloader.koplugin/main.lua [hunk 1]
```diff
@@ -678,7 +678,7 @@ function NewsDownloader:viewFeedList()
         }
     )
     -- Show the list of feeds.
-    if #self.kv ~= 0 then
+    if self.kv ~= null and #self.kv ~= 0 then
```
Comment (zwim) [inline plugins/newsdownloader.koplugin/main.lua:681]: ```if self.kv and #self.kv > 0 then```
should do.

Comment (zwim) [reply plugins/newsdownloader.koplugin/main.lua:681]: BTW: `null` is unknown (and German ;-) ), probably you wanted to use `nil`?

**Review by NiLuJe — APPROVED** [review]
Can't hurt ;).

**Comment by hius07** [comment]
Not possible here too?
https://github.com/koreader/koreader/blob/8de652b38569530af4581b78870ff945d95ac57a/plugins/newsdownloader.koplugin/main.lua#L696-L698

**Comment by hius07** [comment]
Do we need this check at all?
https://github.com/koreader/koreader/blob/e72211a5bf2207a990a9203b287d5769f178dfcb/frontend/ui/uimanager.lua#L175-L179

**Comment by NiLuJe** [comment]
> Do we need this check at all?
> 
> https://github.com/koreader/koreader/blob/e72211a5bf2207a990a9203b287d5769f178dfcb/frontend/ui/uimanager.lua#L175-L179

I wondered, but decided to keep it, because, while unlikely, it might catch typos when using forward declarated local widgets.

i.e.,

```
local widget
widget = FancyWidget:new{}
<... code happens>
UIManager:show(wigdet) <-- typo)
```

**Comment by hius07** [comment]
EDIT: I mean a check in the plugin.

**Comment by NiLuJe** [comment]
> EDIT: I mean a check in the plugin.

Oh, right, sorry; I , err, hadn't actually looked at the code inside that branch all too closely ;).

Not familiar with how said widget is built, but, yeah, avoiding the obvious warning from UIManager would make sense (... assuming the possibility of it being nil in the plugin code even makes sense to begin with).

In which case, a simple `nil` check is enough, I don't quite see the interest of the length check there?

**Comment by roygbyte** [comment]
Yeah, I can't speak to the reasoning of doing a length check of that variable. I'm sure there was "a reason" at some point, but I haven't a clue what that reason would have been. 

For me, this is lipstick on a gorilla, anyways. Any changes to avoid crashes in the short term is good. But I wouldn't unpack it much more than that...

File: plugins/newsdownloader.koplugin/main.lua [hunk 2]
```diff
@@ -678,7 +678,7 @@ function NewsDownloader:viewFeedList()
         }
     )
     -- Show the list of feeds.
-    if #self.kv ~= 0 then
+    if self.kv and #self.kv > 0 then
         UIManager:close(self.kv)
     end
     self.kv = KeyValuePage:new{
```
[the remark this answers is not in the document]
Comment (NiLuJe) [reply plugins/newsdownloader.koplugin/main.lua:684]: Yeah, we touched on that a few days ago; I agree that it doesn't make any sense ;).

[the remark this answers is not in the document]
Comment (roygbyte) [reply plugins/newsdownloader.koplugin/main.lua:684]: It shouldn't make sense, it doesn't make sense, it will never make sense.

**Pull request merged**